tensorflow/core/kernels/random_op.cc

```diff
@@ -558,49 +558,34 @@ TF_CALL_int64(REGISTER_INT);
 
 namespace functor {
 
-using namespace cl;
-
 template <class Distribution, bool VariableSamplesPerOutput>
 struct FillPhiloxRandomKernel;
 
 template <class Distribution>
 struct FillPhiloxRandomKernel<Distribution, false> {
   typedef typename Distribution::ResultElementType T;
-  using write_accessor = sycl::accessor<uint8_t, 1, sycl::access::mode::write,
-                                        sycl::access::target::global_buffer>;
+  using write_accessor =
+    cl::sycl::accessor<uint8_t, 1, cl::sycl::access::mode::write,
+                       cl::sycl::access::target::global_buffer>;
 
   FillPhiloxRandomKernel(write_accessor& data, random::PhiloxRandom& gen,
                          Distribution& dist)
       : data_(data), gen_(gen), dist_(dist) {}
 
-  void operator()(sycl::nd_item<1> item) {
-    const size_t kGroupSize = Distribution::kResultElementCount;
+  void operator()(cl::sycl::nd_item<1> item) {
+    constexpr size_t kGroupSize = Distribution::kResultElementCount;
 
     const size_t item_id = item.get_global(0);
-    const size_t total_item_count = item.get_global_range(0);
-    size_t offset = item_id * kGroupSize;
-    gen_.Skip(item_id);
+    const size_t offset = item_id * kGroupSize;
 
     const size_t size = data_.get_size() / sizeof(T);
-    T* data = ConvertToActualTypeSycl(T, data_);
-
-    while (offset + kGroupSize <= size) {
-      const typename Distribution::ResultType samples = dist_(&gen_);
-      for (size_t i = 0; i < kGroupSize; ++i) {
-        data[offset + i] = samples[i];
-      }
-
-      offset += (total_item_count - 1) * kGroupSize;
-      gen_.Skip(total_item_count - 1);
-    }
+    T* data = ConvertToActualTypeSycl(T, data_) + offset;
```

Comment:
Meh. We don't use auto for `ConvertToActualTypeSycl` anywhere else, maybe we could refactor that later if we decide to? I used auto for samples because the return type is a bit long.